cmd/controller/controller.go

```diff
@@ -155,6 +171,83 @@ func (c *Controller) processNextItem() bool {
 	return true
 }
 
+func fetchUrl(reqURL, authHeader string) (*http.Response, error) {
+	req, err := http.NewRequest("GET", reqURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(authHeader) > 0 {
+		req.Header.Set("Authorization", authHeader)
+	}
+	return netClient.Do(req)
```

Comment:
Embedding the httpClient in this way means you can't inject a fake client for testing.  I would rather that this function accepted an `httpClient` arg, and we passed that down the callstack.

Reply:
`netClient` is a global variable. In the tests we can assign a fake http client to netClient variable. The chart-repo tests does so in it's tests https://github.com/kubeapps/kubeapps/blob/master/cmd/chart-repo/utils_test.go#L190-L191

Reply:
updated!